### DISCUSSION TITLE
AWS HTTP error: cURL error 60: SSL: no alternative certificate subject name matches target host name

### DISCUSSION THREAD

**Discussion by apisitas** [aws/aws-sdk-php#2645] [Q&A]
Hi i am Gokart
I am using this code  from stack overflow https://stackoverflow.com/questions/65370758/upload-files-to-oracle-cloud-storage-using-php.
I have founded error when i upload file to my bucket 46355761746511223ggggggg
This is error  response
{
    "success": false,
    "message": "Error executing \"PutObject\" on \"https://46355761746511223ggggggg.xxx.compat.objectstorage.eu-frankfurt-1.oraclecloud.com/46355761746511223ggggggg/ST-10.png\"; AWS HTTP error: cURL error 60: SSL: no alternative certificate subject name matches target host name '[46355761746511223ggggggg.xxx.compat.objectstorage.eu-frankfurt-1.oraclecloud.com](http://46355761746511223ggggggg.xxx.compat.objectstorage.eu-frankfurt-1.oraclecloud.com/)' (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://46355761746511223ggggggg.xxx.compat.objectstorage.eu-frankfurt-1.oraclecloud.com/46355761746511223ggggggg/ST-10.png"
}
But if i upload file to bucket 46355761746511223ggggggG is working fine.
Seems like it will working only bucket have capital letter text.
This is success response
{
    "success": true,
    "message": "https://objectstorage.eu-frankfurt-1.oraclecloud.com/n/xxx/b/46355761746xxxxxxggggggG/o/ST-10.png"
}

**Comment by yenfryherrerafeliz** [comment]
Hi @apisitas, this error seems to be related to your certificate configuration rather than a problem with the SDK. This error tends to occur when the SSL certificate is not configured correctly or maybe there is a mismatch in between the certificate and the domain name that you are using here and that is giving you the error.

Thanks!

**Comment by Mr-Anonymous** [comment]
Did you find a solution for this error? I have the exact same error happening too. It is in my production server that been linked to S3 bucket for a long time where it uploads the files to the bucket or fetches files from bucket to list. But all of a sudden, today my error log is filled with errors like this:

```
Next Aws\S3\Exception\S3Exception: Error executing "ListObjects" on "https://bucketname.s3.ap-southeast-2.amazonaws.com/?prefix=uploads%2Fapps%2F230%2Fvideos%2Fvideo-jhwuewe.png%2F&max-keys=1&encoding-type=url"; AWS HTTP error: cURL error 60: SSL: no alternative certificate subject name matches target host name 'bucketname.s3.ap-southeast-2.amazonaws.com'
```

What could have changed suddenly? Does this mean the error is probably due to the SSL certificate on my server? Or the certificate from Amazon side? Or has something else changed and my sdk is no longer compatible?

I am using laravel 5.3. and php 7.1.

Really appreciate any assistance you can give.

**Reply by Mr-Anonymous** [reply]
This seems to be an issue from the Amazon side for me. It fixed itself after a few hours.

**Comment by Moongazer** [comment]
We have the same problem which started somewhere in 2023. It only happens occasionally, so most requests going through fine, but in our error logs randomly (every 2-3 days, sometimes multiple times per day) this error is thrown:

`AWS HTTP error: cURL error 60: SSL: no alternative certificate subject name matches target host name 'xxx.s3.eu-central-1.amazonaws.com' (see https://curl.haxx.se/libcurl/c/libcurl-errors.html) for https://xxx.s3.eu-central-1.amazonaws.com/?list-type=2&prefix=a-folder-name%2F&max-keys=1 | Aws\S3\Exception\S3Exception thrown in file /www/htdocs/vendor/aws/aws-sdk-php/src/WrappedHttpHandler.php in line 196`

Current `aws/aws-sdk-php` version is v3.305.2, but it happened also with older versions. PHP v8.1 on Apache2.

**Reply by lajos-resnweb** [reply]
Same here. Sometimes it works, sometimes we got this error on local dev machine.
